docs/webapp.adoc

```diff
@@ -1,30 +1,625 @@
 
-== Blabla
+.webapp.es6:
+[source,javascript,options="nowrap"]
+----
+import thymeleaf from '/lib/thymeleaf';
+const view = resolve('webapp.html');
+
+exports.get = () => ({
+        contentType: 'text/html',
+        body: thymeleaf.render(view, {})
+});
+----
 
 {zwsp} +
 
-=== bleh bleh
+[[script]]
+=== The script
+
+Finally, the script that's called at the end of the HTML.
+
+If you've been through the lesson in the <<guillotine#, previous chapter>>, you might recognize that these functions are mostly the same code as was used in that chapter, just copied into one asset (if you haven't, just see that chapter for reference).
+
+The main function is `requestAndRenderMovies`. It gets its input values from the `MOVIE_LIST_PARAMS` object we defined in the global namespace in the HTML earlier, then uses these to request data about 3 (`movieCount`) movies (`movieType`) under the _movielist_ site (`parentPath`), from the guillotine API. Just like in the previous chapter, the guillotine query string for fetching movies is built with a function, `buildQueryListMovies`. The returned data is parsed into a JSON array of movie objects (`extractToMovieArray`) and passed to the `renderMovie` function, where it's used in a `props` object alongside other values from `MOVIE_LIST_PARAMS`. Along with the `props`, the _MovieList_ entry (`React4xp['MovieList]`) is used to create a renderable react component that is rendered into the target `movieListContainer` element in the DOM with `ReactDOM.render`, now as a top-level react app.
+
+
+.script.es6:
+[source,javascript,options="nowrap"]
+----
+const buildQueryListMovies = (movieType, parentPath) => {
+    const matched = movieType.match(/(\w+(\.\w+)*):(\w+)/i);  // verifies content type names like "com.enonic.app.react4xp:movie" and matches up groups before and after the colon
+    if (!matched) {
+        throw Error(`movieType '${movieType}' is not a valid format. Expecting <appName>:<XP content type>, for example: 'com.enonic.app.react4xp:movie' etc`);
+    }
+    const appNameUnderscored = matched[1].replace(/\./g, '_');      // e.g. "com.enonic.app.react4xp" --> "com_enonic_app_react4xp
+    const ctyCapitalized = matched[3][0].toUpperCase() + matched[3].substr(1);       // e.g. "movie" --> "Movie"
+
+    return `
+query($first:Int!, $offset:Int!, $sort:String!) {
+  guillotine {
+    query(contentTypes: ["${movieType}"], query: "_parentPath = '/content${parentPath}'", first: $first, offset: $offset, sort: $sort) {
+      ... on ${appNameUnderscored}_${ctyCapitalized} {
+        _id
+        displayName
+        data {
+          year
+          description
+          actor
+          image {
+            ... on media_Image {
+              imageUrl(type: absolute, scale: "width(300)")
+            }
+          }
+        }
+      }
+    }
+  }
+}`;
+};
+
+// Not using util-lib to ensure usability on frontend
+const forceArray = maybeArray => Array.isArray(maybeArray)
+    ? maybeArray
```

Comment:
Don't nest lambda please. 🤕

Reply:
See other comment: connected to some rearrangement of code, will take a little bit of time, so I suggest releasing the docs without this improvement and making a task for it.

Reply:
❕

Reply:
- [ ] https://github.com/enonic/doc-react4xp/issues/8